core/client/src/main/java/alluxio/client/file/UnderFileSystemFileInStream.java

```diff
@@ -164,7 +164,10 @@ private ByteBuffer allocateBuffer() {
   private void bufferedRead(int len) throws IOException {
     mBuffer.clear();
     int bytesRead = directRead(mBuffer.array(), 0, len);
```

Comment:
also I feel in that case, we may also pass mReader.read a bytebuffer to write down the data, and save one memory copy? --- this could be a marginal optimization, feel free to not address it in this PR.